source/Octopus.Tentacle/Services/Scripts/ScriptServiceV3Alpha/ScriptServiceV3Alpha.cs

```diff
@@ -0,0 +1,200 @@
+                        command.IsolationMutexName,
+                        command.Arguments,
+                        command.Files);
+
+                    runningScript.ScriptStateStore.Create();
+                }
+
+                var process = LaunchShell(command.ScriptTicket, command.TaskId, workspace, runningScript.ScriptStateStore, runningScript.CancellationToken);
+
+                runningScript.Process = process;
+
+                if (command.DurationToWaitForScriptToFinish != null)
+                {
+                    var waited = Stopwatch.StartNew();
+                    while (process.State != ProcessState.Complete && waited.Elapsed < command.DurationToWaitForScriptToFinish.Value)
+                    {
+                        Thread.Sleep(TimeSpan.FromMilliseconds(10));
+                    }
+                }
+
+                return await GetResponse(command.ScriptTicket, 0, runningScript.Process);
+            }
+        }
+
+        public async Task<ScriptStatusResponseV3Alpha> GetStatusAsync(ScriptStatusRequestV3Alpha request, CancellationToken cancellationToken)
+        {
+            runningScripts.TryGetValue(request.ScriptTicket, out var runningScript);
+            return await GetResponse(request.ScriptTicket, request.LastLogSequence, runningScript?.Process);
+        }
+
+        public async Task<ScriptStatusResponseV3Alpha> CancelScriptAsync(CancelScriptCommandV3Alpha command, CancellationToken cancellationToken)
+        {
+            if (runningScripts.TryGetValue(command.ScriptTicket, out var runningScript))
+            {
+                runningScript.Cancel();
+            }
+
+            return await GetResponse(command.ScriptTicket, command.LastLogSequence, runningScript?.Process);
+        }
+
+        public async Task CompleteScriptAsync(CompleteScriptCommandV3Alpha command, CancellationToken cancellationToken)
+        {
+            if (runningScripts.TryRemove(command.ScriptTicket, out var runningScript))
+            {
+                runningScript.Dispose();
+            }
+
+            var workspace = workspaceFactory.GetWorkspace(command.ScriptTicket);
+            await workspace.Delete(cancellationToken);
+        }
+
+        RunningScript LaunchShell(ScriptTicket ticket, string serverTaskId, IScriptWorkspace workspace, IScriptStateStore stateStore, CancellationToken cancellationToken)
+        {
+            var runningScript = new RunningScript(shell, workspace, stateStore, workspace.CreateLog(), serverTaskId, cancellationToken, log);
+            var thread = new Thread(runningScript.Execute) { Name = "Executing PowerShell runningScript for " + ticket.TaskId };
+            thread.Start();
+            return runningScript;
+        }
+
+        async Task<ScriptStatusResponseV3Alpha> GetResponse(ScriptTicket ticket, long lastLogSequence, RunningScript? runningScript)
+        {
+            await Task.CompletedTask;
```

Comment:
Made this async as it's called in a bunch of places and it _will_ be async in the future :)